src/cargo/ops/cargo_compile.rs

```diff
@@ -370,8 +372,8 @@ pub fn create_bcx<'a, 'cfg>(
     let target_data = RustcTargetData::new(ws, &build_config.requested_kinds)?;
 
     let all_packages = &Packages::All;
-    let rustdoc_scrape_examples = &config.cli_unstable().rustdoc_scrape_examples;
-    let need_reverse_dependencies = rustdoc_scrape_examples.is_some();
+    let need_reverse_dependencies =
+        matches!(build_config.mode, CompileMode::Doc { .. }) && config.nightly_features_allowed;
```

Comment:
There are a lot of local variables in this function, and I'm having a hard time keeping them straight. (spec, specs, full_specs, resolve_specs, need_reverse_dependencies).

Might I suggest a few things to maybe clarify it a little? Maybe something like:

```rust
let needs_scrape_examples =
    matches!(build_config.mode, CompileMode::Doc { .. }) && config.nightly_features_allowed;
let requested_specs = spec.to_package_id_specs(ws)?;
let specs_to_resolve = if needs_scrape_examples {
    // When scraping examples, this neads All because …
    Packages::All.to_package_id_specs(ws)?
} else {
    requested_specs.clone()
};
```

Where this:
* `need_reverse_dependencies` → `needs_scrape_examples`
* `resolve_specs` → `specs_to_resolve`
* Remove `full_specs` and `specs`
* Add `requested_specs` for the specs requested by the user

Also, I'm a bit surprised that scraping forces scraping from the entire workspace. That seems like it could cause a lot of problems. Am I understanding what this is doing correctly?

Reply:
I tried to clean up the code roughly in line with your suggestion. Let me know if you think it's clearer.

And yes, scraping does by default look through all the packages in your workspace for examples. What kind of problems are you anticipating?

Reply:
I'm pretty sure that isn't something we can do by default. Workspaces will have packages that don't compile, or only build on certain targets, or have very expensive build scripts, or any number of requirements. Some workspaces may be huge, but the user only wants to document one crate. There are some users who have monolithic workspaces across entire orgs.

I think if the user wants to document the entire workspace, they can specify `--workspace` or multiple `-p` flags.

Reply:
Ok, that makes sense. I changed it to just use the package set requested by the user, and we can eventually document that users should do `--all` or whatever if they want to scrape all possible examples.